x-pack/plugins/maps/public/components/validated_number_input.tsx

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { Component, ChangeEvent } from 'react';
+import { EuiFieldNumber, EuiFormRow } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import _ from 'lodash';
+
+interface State {
+  value: number | string;
+  errorMessage: string;
+  isValid: boolean;
+}
+
+interface Props {
+  initialValue: number;
+  min: number;
+  max: number;
+  onChange: (value: number) => void;
+  label: string;
+}
+
+function getErrorMessage(min: number, max: number): string {
+  return i18n.translate('xpack.maps.validatedNumberInput.invalidClampErrorMessage', {
+    defaultMessage: 'Must be between {min} and {max}',
+    values: {
+      min,
+      max,
+    },
+  });
+}
+
+function isNumberValid(value: number | string, min: number, max: number) {
+  const parsedValue = parseFloat(value.toString());
+
+  if (isNaN(parsedValue)) {
+    return {
+      isValid: false,
+      errorMessage: getErrorMessage(min, max),
+    };
+  }
+
+  const isValid = parsedValue >= min && parsedValue <= max;
+  return {
+    parsedValue,
+    isValid,
+    errorMessage: isValid ? '' : getErrorMessage(min, max),
+  };
+}
+
+export class ValidatedNumberInput extends Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+
+    const { isValid, errorMessage } = isNumberValid(
+      props.initialValue,
+      this.props.min,
+      this.props.max
+    );
+
+    this.state = {
+      value: props.initialValue,
+      errorMessage,
+      isValid,
+    };
+  }
+
+  _submit = _.debounce((value) => {
+    this.props.onChange(value);
+  }, 250);
+
+  _onChange = (e: ChangeEvent<HTMLInputElement>) => {
+    const value = (e.target as HTMLInputElement).value;
+    const { isValid, errorMessage, parsedValue } = isNumberValid(
+      value,
+      this.props.min,
+      this.props.max
+    );
+
+    this.setState({
+      value,
+      errorMessage,
+      isValid,
+    });
+
+    if (isValid) {
+      this._submit(parsedValue);
+    }
+  };
+
+  render() {
+    return (
+      <EuiFormRow
+        label={this.props.label}
+        isInvalid={!this.state.isValid}
+        error={this.state.errorMessage ? [this.state.errorMessage] : []}
+        display="columnCompressed"
+      >
+        <EuiFieldNumber
+          min={this.props.min}
```

Comment:
Pass isInvalid to EuiFieldNumber so it shows error state. Notice in the screen shot below how there is red line under the missing field name select but there is a blue line under the number input even though its in error state.

<img width="300" alt="Screen Shot 2020-12-15 at 8 34 12 AM" src="https://user-images.githubusercontent.com/373691/102236906-499b9a80-3eb1-11eb-8c6f-bbfb6366f54d.png">